backend/tests/v5/unit/services/filesManager.test.js

```diff
@@ -432,7 +433,65 @@ const testFileExists = () => {
 	});
 };
 
+const testRemoveFilesWithMeta = () => {
+	describe('Remove files with meta', () => {
+		const meta = { [generateRandomString()]: generateRandomString() };
+		const teamspace = generateRandomString();
+		const collection = generateRandomString();
+		test('Should remove all files that satisfies the query', async () => {
+			const expectedRefs = times(3, () => ({ _id: generateRandomString(), type: 'fs', link: generateRandomString() }));
+			FileRefs.getRefsByQuery.mockResolvedValueOnce(expectedRefs);
+
+			await FilesManager.removeFilesWithMeta(teamspace, collection, meta);
+
+			expect(FileRefs.getRefsByQuery).toHaveBeenCalledTimes(1);
+			expect(FileRefs.getRefsByQuery).toHaveBeenCalledWith(teamspace, collection, meta);
+
+			expect(FSHandler.removeFiles).toHaveBeenCalledTimes(1);
+			expect(FSHandler.removeFiles).toHaveBeenCalledWith(expectedRefs.map(({ link }) => link));
+
+			expect(FileRefs.removeRefsByQuery).toHaveBeenCalledTimes(1);
+			expect(FileRefs.removeRefsByQuery).toHaveBeenCalledWith(teamspace, collection, meta);
+		});
+
+		test('Should do nothing if no files are found', async () => {
+			FileRefs.getRefsByQuery.mockResolvedValueOnce([]);
+
+			await FilesManager.removeFilesWithMeta(teamspace, collection, meta);
+
+			expect(FileRefs.getRefsByQuery).toHaveBeenCalledTimes(1);
+			expect(FileRefs.getRefsByQuery).toHaveBeenCalledWith(teamspace, collection, meta);
+
+			expect(FSHandler.removeFiles).not.toHaveBeenCalled();
+			expect(FileRefs.removeRefsByQuery).not.toHaveBeenCalled();
+		});
+
+		test('Should manage not file refs properly', async () => {
```

Comment:
should manage * ?

Reply:
non file refs -t his is a test to make sure we don't try to delete a file with http type (.i.e. a hyperlink)